astropy/coordinates/tests/test_frames.py

```diff
@@ -1676,3 +1676,48 @@ def test_frame_coord_comparison():
     error_msg = "Can only compare SkyCoord to Frame with data"
     with pytest.raises(ValueError, match=error_msg):
         frame == coord  # noqa: B015
+
+
+@pytest.mark.parametrize(
+    ["s1", "s2"],
+    (
+        ((1,), (1,)),
+        ((2,), (1,)),
+        ((1,), (2,)),
+        ((2,), (2,)),
+        ((2, 1), (1,)),
+        ((1,), (2, 1)),
+        ((2, 1), (1, 3)),
+    ),
+)
+def test_altaz_broadcast(s1, s2):
+    """Note: Regression test for #5982"""
+    where = EarthLocation.from_geodetic(lat=45 * u.deg, lon=30 * u.deg, height=0 * u.m)
+    time = Time(np.full(np.prod(s1), 58000.0).reshape(s1), format="mjd")
+    angle = np.full(np.prod(s2), 45.0).reshape(s2) * u.deg
+    AltAz(alt=angle, az=angle, obstime=time, location=where)
```

Comment:
I used [np.broadcast_shapes](https://numpy.org/doc/stable/reference/generated/numpy.broadcast_shapes.html#numpy.broadcast_shapes), which is new in Numpy 1.20.0.

Reply:
Yes, nice. Also suggests we can deprecate `check_broadcast`...

Reply:
I agree. I had started working on a patch for that, but I quickly realized that it would hit most of the same lines as this PR. We can come back to that in a subsequent PR. I opened #15211 to remind us.